### PR TITLE
Clarify 'anyOf' errors

### PR DESCRIPTION
**Pull request by mrenvoize**
This pull request goes some way towards clarifying anyOf error messages.. It's limits us down to the single error at the deepest failure point in the tree.

### CODE REVIEW

**Comment by jhthorsen** [comment]
Sorry, I don't get why this change makes the error more clear. Maybe it will be more obvious if you provide a more complex test in addition to changing the existing tests.

**Comment by mrenvoize** [comment]
Sure thing,.. I'll do a more in depth test shortly..

But.. in short.. for nested errors we go from:

{ path: '/', message: "anyOf failed: anyOf failed: anyOf failed: Expected String got Number" }

to:

{ path:'/a/b/c/d', message: "Expected String got Number" }

**Comment by jhthorsen** [comment]
That sounds fantastic! Please prove it with a test, so we don't go back to the confusing message.

**Comment by mrenvoize** [comment]
At last, I think I've got there..

It definitely needs a bit of golfing though.. the code is somewhat cringy in places but the functionality is working.

File: t/jv-anyof.t [hunk 1]
```diff
@@ -80,7 +80,8 @@ my $schemaC = {
 
 @errors = $validator->validate("string not integer", $schemaC);
 
-is "@errors", "/: anyOf failed: Expected object or integer, got string.", "nesting: string not integer (or object)";
+is "@errors", "/: anyOf failed: Expected integer, got string.", "nesting: string not integer (or object)";
+# It would be nice to get 'object or integer' in the error above
```
Comment (jhthorsen) [inline t/jv-anyof.t]: I think avoiding "object" makes it clearer.

Comment (mrenvoize) [inline t/jv-anyof.t]: You could be right.. I dunno.. happy for you to make that call

Comment (mrenvoize) [inline t/jv-anyof.t]: Want me to remove it?

**Comment by jhthorsen** [comment]
Closing this because of inactivity.

**Pull request closed without merging**